### DISCUSSION TITLE
Verification typing

### DISCUSSION THREAD

**Discussion by m-carrasco** [jbevain/cecil#845] [Q&A] [answered]
Hi,

Do you know of any verification typing algorithm for CIL that can be used as a library? I'd like to know the verification type of each stack value at every instruction. 

Worst-case scenario, I would implement one myself using Cecil. Have you ever tried something like this? Any piece of advice is more than welcomed!

Best regards,
Manuel.

**Comment by jbevain** [comment, marked as answer]
Hi, Unfortunately I'm not aware of one :) Good luck!
